### DISCUSSION TITLE
Integrating with the Popover API

### DISCUSSION THREAD

**Discussion by AndrewKGuan** [floating-ui/floating-ui#2196] [Ideas]
Hi,
I'm wondering if you would be open to the idea of leveraging this native API supported by some of the browsers to take advantage of its performance:
https://developer.chrome.com/docs/web-platform/popover-api/

**Comment by atomiks** [comment]
Hi, can you explain what that might look like? How would it be integrated with the library?

**Reply by FezVrasta** [reply]
The fact they are always positioned outside the document flow may have perf implications

**Reply by AndrewKGuan** [reply]
@atomiks hey, sure, so there are 2 ways, I think, that the API can be integrated: 

- detect browser support and call the API if yes, and fall back to existing logic if no.
- there'll be a polyfill coming soon from the same folks who are working on the API, it'll take care of feature detection.  You could swap that in under your library's code base 

@FezVrasta the browser API is designed to do things at the browser level to provide much faster performance than the JS counterpart.  If the browser supports it, the only JS logic needed is feature detection on load, which is much shorter execution path than any JS solutions.

**Reply by FezVrasta** [reply]
The popover API doesn't provide positioning, it only provides a way to render the popover content in the page. It's very similar to the `<dialog>` element

**Reply by AndrewKGuan** [reply]
you could use it in conjunction with [the anchor positioning API](https://drafts.csswg.org/css-anchor-position-1/), it allows calculations of where the popover should be placed. To be fair though, it isn't available for testing on browsers until sometime later this year.  But just wanted to put this idea out there.

**Reply by FezVrasta** [reply]
We already evaluated the anchor positioning API, is quite limited so we doubt it'll get to replace all the Floating UI functionalities.

**Reply by AndrewKGuan** [reply]
that's good to hear, thanks!
Am just curious, do you mind telling me what those functionalities are that the API is limited on?

**Reply by FezVrasta** [reply]
You can find some info here https://github.com/floating-ui/floating-ui/discussions/2074

**Reply by AndrewKGuan** [reply]
Thanks! I appreciate it.
